test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -82,6 +84,42 @@ func AutomationConfigConfigMapExists(mdb *mdbv1.MongoDB) func(t *testing.T) {
 	}
 }
 
+// FeatureCompatibilityVersion verifies that the FeatureCompatibilityVersion is
+// set to `version`. The FCV parameter is not signaled as a non Running state, for
+// this reason, this function checks the value of the parameter many times, based
+// on the value of `tries`.
+func FeatureCompatibilityVersion(mdb *mdbv1.MongoDB, version string, tries int) func(t *testing.T) {
+	return func(t *testing.T) {
+		assert.GreaterOrEqual(t, tries, 0)
+
+		ctx, _ := context.WithTimeout(context.Background(), 10*time.Minute)
+		mongoClient, err := mongo.Connect(ctx, options.Client().ApplyURI(mdb.MongoURI()))
+		assert.NoError(t, err)
+
+		database := mongoClient.Database("admin")
+		assert.NotNil(t, database)
+
+		found := false
+		for i := 0; i < tries; i++ {
+			runCommand := bson.D{primitive.E{Key: "getParameter", Value: 1}, primitive.E{Key: "featureCompatibilityVersion", Value: 1}}
+			var result bson.M
+			err = database.RunCommand(ctx, runCommand).Decode(&result)
+			assert.NoError(t, err)
+
+			expected := primitive.M{"version": version}
+			if reflect.DeepEqual(expected, result["featureCompatibilityVersion"]) {
+				found = true
+				break
+			}
+
+			// Give the MongoDB server a few seconds to reach the expected FCV.
+			time.Sleep(10)
```

Comment:
instead of dealing a loop with time.Sleep, we could leverage the use of contexts and `time.After` https://www.sohamkamani.com/golang/2018-06-17-golang-using-context-cancellation/